h2o-algos/src/test/java/hex/tree/PrintMojoTreeTest.java

```diff
@@ -0,0 +1,198 @@
+    System.setSecurityManager(new PreventExitSecurityManager());
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    System.setSecurityManager(originalSecurityManager);
+  }
+
+  @Test
+  public void testMojoCategoricalPrint() throws IOException {
+    try {
+      Scope.enter();
+      Frame train = Scope.track(TestUtil.parse_test_file("smalldata/iris/iris.csv"));
+
+      IsolationForestModel.IsolationForestParameters p = new IsolationForestModel.IsolationForestParameters();
+      p._train = train._key;
+      p._ignored_columns = new String[]{"C1", "C2", "C3", "C4"};
+      p._seed = 0xFEED;
+      p._ntrees = 1;
+
+      IsolationForestModel model = new IsolationForest(p).trainModel().get();
+      final File modelFile = folder.newFile();
+      model.exportMojo(modelFile.getAbsolutePath(), true);
+
+      final File treeOutput = folder.newFile();
+      try {
+        PrintMojo.main(new String[]{"--input", modelFile.getAbsolutePath(), "--output", treeOutput.getAbsolutePath()});
+        fail("Expected PrintMojo to call System.exit()");
+      } catch (PreventedExitException e) {
+      }
+
+      final String treeDotz = FileUtils.readFileToString(treeOutput);
+      System.out.println(treeDotz);
+      assertFalse(treeDotz.isEmpty());
+
+
+      final Pattern labelPattern = Pattern.compile("label{1}=\\\"(.*?)\\\"");
+      final Pattern labelContentPattern = Pattern.compile(".*[<>=].*"); // Contains < or > or =
+      final Matcher matcher = labelPattern.matcher(treeDotz);
+
+      assertEquals(1, matcher.groupCount());
+
+      while (matcher.find()){
+        assertFalse(labelContentPattern.matcher(matcher.group(1)).matches());
+      }
+
+    } finally {
+      Scope.exit();
+    }
+  }
+
+  @Test
+  public void testMojoCategoricalPrint_limitedLevels() throws IOException {
+    try {
+      Scope.enter();
+      Frame train = Scope.track(TestUtil.parse_test_file("smalldata/iris/iris.csv"));
+
+      IsolationForestModel.IsolationForestParameters p = new IsolationForestModel.IsolationForestParameters();
+      p._train = train._key;
+      p._ignored_columns = new String[]{"C1", "C2", "C3", "C4"};
+      p._seed = 0xFEED;
+      p._ntrees = 1;
+
+      IsolationForestModel model = new IsolationForest(p).trainModel().get();
+      final File modelFile = folder.newFile();
+      model.exportMojo(modelFile.getAbsolutePath(), true);
+
+      final File treeOutput = folder.newFile();
+      try {
+        PrintMojo.main(new String[]{"--input", modelFile.getAbsolutePath(), "--output", treeOutput.getAbsolutePath(), "--levels", "1"});
+        fail("Expected PrintMojo to call System.exit()");
+      } catch (PreventedExitException e) {
+      }
+
+      final String treeDotz = FileUtils.readFileToString(treeOutput);
+      assertFalse(treeDotz.isEmpty());
+
+      assertTrue(treeDotz.contains("\"SG_0_Node_0\" -> \"SG_0_Node_1\" [fontsize=14, label=\"[NA]\\n2 levels\\n\"]\n" +
```

Comment:
Hmmm, doesn't this part rely too much on the output ? The implementation can change. However, this is the most precise control.